### DISCUSSION TITLE
supabase-js: joining on a foreign key from a second table

### DISCUSSION THREAD

**Discussion by brydonm** [supabase/supabase#4680] [Questions] [answered]
Hey everyone. I'm trying to query some data by joining on an ID from the second table and having some trouble finding the syntax for Supabase.

```md
wars
- id int8
- type varchar
- ...

war_players
- war_id int8
- ...
```

I'm trying to do something like this but with a Supabase JS query:

```sql
SELECT *
FROM wars
LEFT OUTER JOIN war_players ON war_players.war_id = wars.id
```

Here's my intended returned value from Supabase:
```json
[
  { // war
    "id": 1
    "type": "war",
    "war_players": [
      { // war_player
        "id": 123,
        "war_id": 1
      }
    ]
  }
]
```

My initial thought was something like the below code would work, but it did not as I'm guessing it assumes the `war` has a connection to a `war_player` whereas it's the opposite

```js
supabase
      .from("wars")
      .select(`
        id,
        type,
        war_players:war_id(
          name
        )
      `)
```

Any help is appreciated, thank you!

**Comment by steve-chavez** [comment, marked as answer]
What's the result you get for the above snippet?

Also, can you try it as:

```js
supabase
      .from("wars")
      .select(`
        id,
        type,
        war_players(
          name
        )
      `)
```

**Reply by brydonm** [reply]
Well, I was not imagining it to be this simple. How amazing, thank you @steve-chavez! The snippet I posted above caused an error because there was no `war_id` within `wars`. There is no relation from `wars` -> `war_players` but there is from `war_players` -> `wars` and I guess the SDK is smart enough to know what the result is with your snippet.

**Reply by Polyterative** [reply]
@Brydom  Can you please provide the code you used in the end? I'm facing a similar problem

**Reply by brydonm** [reply]
@Polyterative here you go!

```js
    const warsResponse = await supabase
      .from("wars")
      .select(
        `
      id,
      ...
  `
      )
      .contains("roster", JSON.stringify([{ id: user.id }]));
```

The `roster` is set up as a `jsonb` type, not `jsonb[]` and here is an example of the value:
```json
[
  {
    "id": "uuid",
    "name": "Yvor"
  }
]
```
